Make pytest ending output less chatty

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82262

If you still want this in CI, we should have a separate CI only
configuration.  The current config is pretty unfriendly for local
development.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
Make pytest ending output less chatty

If you still want this in CI, we should have a separate CI only
configuration.  The current config is pretty unfriendly for local
development.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

clee2000 commented on pytest.ini:
```diff
@@ -1,13 +1,12 @@
 addopts =
     # show summary of all tests that did not pass
-    -ra
+    -rf
```
i think this would actually have to happen in `common_utils.py`, specifically get rid of the `s`

also note that skip reasons would no longer be printed

clee2000 replied:
ok this is dumb, but if you change `-rfEsX` in `common_utils.py` to `-rsfEX` it might show the skips before the fails -> the fails are closer to the bottom -> scroll less, but you'd still need to take the test name and control f for the stacktrace

i have a change in test-infra for the hud log viewer to go to the stacktraces but when the hud log viewer is minimized its going to look like its just pointing at a line that says `==== FAILURES =====` so youd have to expand to see the name of the failing test and its stacktrace

ezyang replied:
Is there a reason we want to show skips and xfails? It doesn't seem like they need to be in the short summary.

clee2000 replied:
skips because thats the only place skip reasons are printed, the X is for xpassed

ezyang replied:
are we using the skip reasons for anything though? 🤔

ezyang replied:
OK so I double checked and when I invoke pytest locally, it is only pytest.ini that is respected. So if we can't agree on how CI should work, I would still like to get the pytest.ini portion of this change in.

ezyang committed:
Update on "Make pytest ending output less chatty"


If you still want this in CI, we should have a separate CI only
configuration.  The current config is pretty unfriendly for local
development.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Make pytest ending output less chatty"


If you still want this in CI, we should have a separate CI only
configuration.  The current config is pretty unfriendly for local
development.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

ezyang commented:
@pytorchbot rebase

ezyang committed:
Update on "Make pytest ending output less chatty"


If you still want this in CI, we should have a separate CI only
configuration.  The current config is pretty unfriendly for local
development.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

ezyang commented:
@pytorchbot rebase

ezyang committed:
Update on "Make pytest ending output less chatty"


If you still want this in CI, we should have a separate CI only
configuration.  The current config is pretty unfriendly for local
development.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge -g

ezyang commented:
@pytorchbot merge -g

ezyang commented:
@pytorchbot merge -f "known master breakage"
